packages/backend/src/document.rs

```diff
@@ -349,3 +374,59 @@ pub async fn search_ref_stubs(
         items,
     })
 }
+
+pub async fn get_ref_children_stubs(ctx: AppCtx, ref_id: Uuid) -> Result<Vec<RefStub>, AppError> {
+    let user_id = ctx.user.as_ref().map(|u| u.user_id.clone());
+
+    let stub_rows = sqlx::query!(
+        r#"
+        WITH child_refs AS (
+            SELECT ARRAY_AGG(refs.id) AS child_ids
+            FROM refs
+            JOIN snapshots ON snapshots.id = refs.head
+            WHERE (
+                get_max_permission($2, refs.id) >= 'read'
+                AND jsonb_path_exists(
+                    snapshots.content,
+                    '$.*[*] ? (@._id == $id && (@.type == "diagram-in" || @.type == "analysis-of"))',
```

Comment:
Playing around in a JSON Path editor, I believe that the first part of this query, namel

```
$.*[*] ? (@._id == $id)
```

will find any top-level object value containing an `_id` key with the given value. Is that right? If so, great. Can you then:

- add a comment or docstring explaining that
- remove the clause `(@.type == "diagram-in" || @.type == "analysis-of")`: for now I would rather be under-specific than over-specific and it's nice to avoid hard-coding these types in the query